### PR TITLE
Transformations: State feature (alpha/beta) and more

### PR DESCRIPTION
**Pull request by torkelo**
* [x] Use Card component from @grafana/ui 
* [x] Refactor and unify plugin state display / badge into a single component 
* [x] Deprete AlphaNotice (was ever only used from DataSourceSettingsPage) 
* [x] State feature to show transforms as being in beta state. 
* [x] Add help / info function to transform row that when clicked would show an info alert with more description and link to docs. Similar to query editor help button
* [x] New HelpBox component as the current Alert does not work well with large markdown content  [figma](https://www.figma.com/file/HpUCIYdxliPC9N7AABkX3X/Large-info-boxes?node-id=0%3A1)
* [x] Help text markdown or fallback to the description.  All end with a link to transformation root docs. 
* [x] Sort transform list in alphabetical order 

Follow-up PR
* Disable / enable transform
* Disable all transforms

Would like to introduce an "experimental" or "preview" state. Alpha needs feature to be enabled, beta = signals feature complete but buggy. Would like something in the middle that would not require a feature toggle but still experimental. Thoughts?

beta flag
Screenshot from 2021-07-11 20-25-25

beta flag and help icon button in header row
Screenshot from 2021-07-11 20-25-58

Help opened  (Just test will remove this unrelated help text before merge, just to test)
Screenshot from 2021-07-11 20-26-13

### CODE REVIEW

**Comment by dprokop** [comment]
@torkelo I think this is a great addition!

> image

Don't like how help text is disconnected from the transformation row header. Maybe we could skip the "Transformation help" sub header and move the help box few pxls up so that it's "connected" with the transformation row header?

**Comment by torkelo** [comment]
@dprokop it's a good idea to test, but not working great, makes the help box less distinct and harder to understand what is going on / how to close it.  
Screenshot from 2021-07-12 09-40-11

**Comment by dprokop** [comment]
@torkelo I think it would work better if there wasn't the "Transformation help" header

**Comment by torkelo** [comment]
@dprokop that is true! it looks great but no clear way on how to close i

Think if we add a color active state to help button. And maybe to close it via the bottom border? 
Screenshot from 2021-07-12 09-51-17

**Comment by torkelo** [comment]
Screenshot from 2021-07-12 10-06-36

**Comment by torkelo** [comment]
Screenshot from 2021-07-12 10-45-21
Screenshot from 2021-07-12 10-56-16
Screenshot from 2021-07-12 10-54-06

File: packages/grafana-ui/src/components/AlphaNotice/AlphaNotice.tsx [hunk 1]
```diff
@@ -10,7 +10,9 @@ export interface Props {
   className?: string;
 }
 
+/** @deprecated Will be removed  in a future release */
 export const AlphaNotice: FC<Props> = ({ state, text, className }) => {
```
Comment (dprokop) [inline packages/grafana-ui/src/components/AlphaNotice/AlphaNotice.tsx:14]: Maybe we can just remove it straight away? I doubt it's used externally (maybe enterprise...)

**Review by dprokop — APPROVED** [review]
LGTM

**Pull request merged**